src/core/bpf-firewall.c

```diff
@@ -583,7 +675,8 @@ int bpf_firewall_install(Unit *u) {
         u->ip_bpf_ingress_installed = bpf_program_unref(u->ip_bpf_ingress_installed);
 
         if (u->ip_bpf_egress) {
-                r = bpf_program_cgroup_attach(u->ip_bpf_egress, BPF_CGROUP_INET_EGRESS, path, flags);
+                r = bpf_program_cgroup_attach(u->ip_bpf_egress, BPF_CGROUP_INET_EGRESS, path,
+                                              !set_isempty(u->ip_bpf_custom_egress) ? BPF_F_ALLOW_MULTI : flags);
```

Comment:
it doesn't really matter, but I think it would be nicer to do:

```
…, flags | (set_isempty(u->ip_bpf_custom_egress) ? 0 : BPF_F_ALLOW_MULTI));
```

i.e. always OR in the `flags` field, even though sometimes it's redundant. That way, maybe one day if a new flags value is defined we don't lose it here.

Reply:
Good point!